LoopBackTests/LBPersistedModelSubclassingTests.m

```diff
@@ -157,52 +187,67 @@ - (void)testAll {
-        XCTAssertEqualObjects(widget.bars, @1, @"Invalid bars");
-        XCTAssertEqualObjects(widget.date, [NSDate dateWithTimeIntervalSince1970:0], @"Invalid date");
-
-        widget.name = savedName;
-        widget.date = savedDate;
-        [model saveWithSuccess:^{
-            ASYNC_TEST_SIGNAL
-        } failure:ASYNC_TEST_FAILURE_BLOCK];
-    };
-
-    LBPersistedModelSaveSuccessBlock findAgain = ^() {
-        [self.repository findById:@2 success:verify failure:ASYNC_TEST_FAILURE_BLOCK];
-    };
-
-    LBPersistedModelFindSuccessBlock update = ^(LBPersistedModel *model) {
-        XCTAssertNotNil(model, @"No model found with ID 2");
-        XCTAssertTrue([[model class] isSubclassOfClass:[Widget class]], @"Invalid class.");
-        Widget *widget = (Widget *)model;
-        savedName = widget.name;
-        savedDate = widget.date;
         widget.name = @"Barfoo";
-        widget.date = [NSDate dateWithTimeIntervalSince1970:0];
-        [model saveWithSuccess:findAgain failure:ASYNC_TEST_FAILURE_BLOCK];
-    };
-
-    [self.repository findById:@2 success:update failure:ASYNC_TEST_FAILURE_BLOCK];
+        widget.data = @{@"data3": @3, @"data4": @4};
+        widget.array = @[@"four", @"five", @"six"];
+        widget.date = [NSDate dateWithTimeIntervalSince1970:123];
+        XCTAssertTrue([widget.buffer length] == 3, @"Invalid buffer length.");
+        ((char *)widget.buffer.mutableBytes)[2] = 65;
+        CLLocation *newGeoPoint = [[CLLocation alloc] initWithLatitude:12.3 longitude:45.6];
+        widget.geopoint = newGeoPoint;
+
+        [widget saveWithSuccess:^() {
+            // find again
+            [self.repository findById:@2 success:^(LBPersistedModel *model) {
+                // verify
+                XCTAssertNotNil(model, @"No model found with ID 2");
+                XCTAssertTrue([[model class] isSubclassOfClass:[Widget class]], @"Invalid class.");
+                Widget *widget = (Widget *)model;
+                XCTAssertEqualObjects(widget.name, @"Barfoo", @"Invalid name");
+                XCTAssertEqualObjects(widget.bars, @1, @"Invalid bars");
+                XCTAssertEqualObjects(widget.data, (@{@"data3": @3, @"data4": @4}), @"Invalid data");
+                XCTAssertEqualObjects(widget.array, (@[@"four", @"five", @"six"]), @"Invalid array.");
+                XCTAssertEqualObjects(widget.date, [NSDate dateWithTimeIntervalSince1970:123], @"Invalid date");
+                const char bufferBytes[] = { 12, 34, 65 };
+                NSMutableData *testData = [NSMutableData dataWithBytes:bufferBytes length:sizeof(bufferBytes)];
+                XCTAssertEqualObjects(widget.buffer, testData, @"Invalid buffer.");
+                XCTAssertEqual(widget.geopoint.coordinate.latitude, 12.3, @"Invalid latitude.");
+                XCTAssertEqual(widget.geopoint.coordinate.longitude, 45.6, @"Invalid longitude.");
+
+                // restore
```

Comment:
Good suggestion.  I didn't like the way it is currently handled neither.  I'm thinking to update the test not to depend on a model stored on the server, rather creates a temporary model and update it, so that we don't need to worry about target model's state before running the test.